internal/app/machined/main.go

```diff
@@ -241,6 +242,23 @@ func run() error {
 		log.Printf("controller runtime finished")
 	}()
 
+	var eg errgroup.Group
```

Comment:
should it be `eg.WithContext` [1]? That way context is cancelled and propagated appropriately too

[1] https://pkg.go.dev/golang.org/x/sync/errgroup

Reply:
yeah, that's a good idea, thanks.